src/pages/curriculums/[id].js

```diff
@@ -1,14 +1,36 @@
+import React from 'react';
 import { useRouter } from 'next/router';
+import dynamic from 'next/dynamic';
 import LearningUnitsSection from '@components/learning-units-section/LearningUnitsSection';
 import { Skeleton } from 'primereact/skeleton';
+import { TabView, TabPanel } from 'primereact/tabview';
+
+const Graph = dynamic(() => import('../../components/learning-units-section/Graph'), {
+  ssr: false,
+});
 
 function CurriculumPage() {
   const { query, isReady } = useRouter();
   const curriculumId = query.id;
```

Comment:
Defining curriculumId before checking that the Router is ready (and before the router is actually ready) makes the app to create curriculumID undefined and then to reassign its value, so it's better to define it after the isReady check

Reply:
good catch